src/test/test_vcfreader.cpp

```diff
@@ -0,0 +1,192 @@
+#include "misc.hpp"
+#include "VcfReader.hpp"
+
+#include <iostream>
+
+using std::stoul;
+using std::cerr;
+using std::cout;
+using std::to_string;
+using std::ofstream;
+using sv_merge::run_command;
+using sv_merge::VcfRecord;
+using sv_merge::for_record_in_vcf;
+
+
+/**
+ * Testing ranges
+ */
+const vector<int> MIN_SV_LENGTHS = {0, 50, 100, 500, 1000};
+const vector<float> MIN_QUALS = {0.0, 10.0, 20.0, 30.0};
+const vector<float> MIN_AFS = {0.02, 0.08, 0.16, 0.32, 0.64};
+const vector<float> MIN_NMFS = {0.02, 0.08, 0.16, 0.32, 0.64};
+const vector<string> CHROMOSOMES = {"chr1","chr2","chr3","chr4","chr5","chr6","chr7","chr8","chr9","chr10","chr11","chr12","chr13","chr14","chr15","chr16","chr17","chr18","chr19","chr20","chr21","chr22","chrX","chrY"};
+const vector<string> SAMPLE_IDS = {"NA24385", "HG03125", "HG00512"};
+const string HPRC_FILE_ID = "hprc-v1.1-mc-chm13.raw.sv";
+const uint32_t PROGRESS_EVERY_LINES = 10000;  // Arbitrary
+
+string N_THREADS;
+
+/**
+ * Assumed to contain every raw input file used for testing
+ */
+path ROOT_DIR;
+
+/**
+ * Temporary files created during the tests
+ */
+path input_vcf, test_vcf, truth_vcf, tmp1_vcf, tmp2_vcf;
+ofstream outstream;
```

Comment:
None of these need to be global

Reply:
Done (I think it's not so important here, since this is essentially a glorified bash script. I left them global to avoid passing the same parameters around...)